projects/kit/src/lib/masks/time/utils/stringify-time.ts

```diff
@@ -0,0 +1,45 @@
+import {DEFAULT_TIME_SEGMENT_MAX_VALUES} from '../../../constants';
+import type {MaskitoTimeSegments} from '../../../types';
+import {padTimeSegments} from '../../../utils/time';
+import type {MaskitoTimeParams} from '../maskito-time-params';
+
+export function maskitoStringifyTime(
+    milliseconds: number | null,
+    {
+        mode,
+        timeSegmentMaxValues = {},
+    }: Pick<MaskitoTimeParams, 'mode' | 'timeSegmentMaxValues'>,
+): string {
+    if (milliseconds === null) {
+        return '';
+    }
+
+    const maxValues: MaskitoTimeSegments<number> = {
+        ...DEFAULT_TIME_SEGMENT_MAX_VALUES,
+        ...timeSegmentMaxValues,
+    };
+
+    const millisecondsInSecond = maxValues.milliseconds + 1;
+    const millisecondsInMinute = (maxValues.seconds + 1) * millisecondsInSecond;
+    const millisecondsInHour = (maxValues.minutes + 1) * millisecondsInMinute;
+
+    const hours = Math.trunc(milliseconds / millisecondsInHour);
+
+    milliseconds -= hours * millisecondsInHour;
+
+    const minutes = Math.trunc(milliseconds / millisecondsInMinute);
+
+    milliseconds -= minutes * millisecondsInMinute;
+
+    const seconds = Math.trunc(milliseconds / millisecondsInSecond);
+
+    milliseconds -= seconds * millisecondsInSecond;
+
+    const result = padTimeSegments({hours, minutes, seconds, milliseconds});
+
+    return mode
+        .replaceAll(/H+/g, result.hours)
+        .replaceAll('MSS', result.milliseconds)
+        .replaceAll(/M+/g, result.minutes)
+        .replaceAll(/S+/g, result.seconds);
```

Comment:
Could we use native `Date` anyway ?

```ts
export function maskitoStringifyTime(ms: number): string {
    const date = new Date(ms);
    const hours = date.getUTCHours();
    const minutes = date.getUTCMinutes();
    const seconds = date.getUTCSeconds();
    const milliseconds = date.getUTCMilliseconds();

    // [...]
    // more logic if required

    return `${hours}:${minutes}:${seconds}.${milliseconds}`;
}

maskitoStringifyTime(86399999); // '23:59:59.999'
```

Reply:
We discussed this and came to the conclusion that this would do more harm than good.